### ISSUE TITLE
Possible -1 array index reference in lib/properties/optimizer.js

### ISSUE THREAD

**Issue by rgbohner** [jakubpawlowicz/clean-css#544]
https://github.com/jakubpawlowicz/clean-css/blob/master/lib/properties/optimizer.js starting at line : 97

variable left can potentially reference properties[-1] causing "TypeError: Cannot read property 'all' of undefined" at line 101

```
  function sameValue(position) {
    var left = properties[position - 1];
    var right = properties[position];

    return stringifyProperty(left.all, left.position) == stringifyProperty(right.all, right.position);
  }
```

possibly change to:
```
  function sameValue(position) {
    if(position > 0){
      var left = properties[position - 1];
      var right = properties[position];

      return stringifyProperty(left.all, left.position) == stringifyProperty(right.all, right.position);
    } else {
      return false;
    }
  }
```

**Comment by jakubpawlowicz** [comment]
Do you have an example where it happens?

**Comment by rgbohner** [comment]
Yep its a bit esoteric but I did run into this on my latest build.  

Here is the information:
The css that will cause this error:
```
.btn-danger {
  background-image: -webkit-gradient(linear, left 0%, left 100%, from(#d9534f), to(#c9302c));
  background-image: -webkit-linear-gradient(top, #d9534f 0%, #c9302c 100%);
}

.btn-danger {
  background-image: -webkit-linear-gradient(top, #d9534f 0%, #c12e2a 100%);
}
```

Stack Trace :
```
    at _optimize (/node_modules/grunt-contrib-cssmin/node_modules/clean-css/lib/properties/optimizer.js:127:15)
    at optimize (/node_modules/grunt-contrib-cssmin/node_modules/clean-css/lib/properties/optimizer.js:200:3)
    at AdvancedOptimizer.reduceSelector (/node_modules/grunt-contrib-cssmin/node_modules/clean-css/lib/selectors/optimizers/advanced.js:229:3)
    at AdvancedOptimizer.reduceSimpleNonAdjacentCases (/node_modules/grunt-contrib-cssmin/node_modules/clean-css/lib/selectors/optimizers/advanced.js:146:12)
    at AdvancedOptimizer.reduceNonAdjacent (/node_modules/grunt-contrib-cssmin/node_modules/clean-css/lib/selectors/optimizers/advanced.js:128:8)
    at _optimize (/node_modules/grunt-contrib-cssmin/node_modules/clean-css/lib/selectors/optimizers/advanced.js:753:12)
    at AdvancedOptimizer.optimize (/node_modules/grunt-contrib-cssmin/node_modules/clean-css/lib/selectors/optimizers/advanced.js:774:3)
    at SelectorsOptimizer.process (/node_modules/grunt-contrib-cssmin/node_modules/clean-css/lib/selectors/optimizer.js:18:55)
    at /node_modules/grunt-contrib-cssmin/node_modules/clean-css/lib/clean.js:186:31
    at run (/node_modules/grunt-contrib-cssmin/node_modules/clean-css/lib/clean.js:172:7)
```

Data on error
```
_name = background-image lastName = background-image
properties[0] = 
{
    "components": [],
    "dirty": false,
    "hack": false,
    "important": false,
    "name": "background-image",
    "multiplex": false,
    "position": 0,
    "shorthand": false,
    "unused": false,
    "value": [
        [
            "-webkit-gradient(linear,left 0,left 100%,from(#d9534f),to(#c9302c))"
        ]
    ],
    "all": [
        [
            [
                "background-image",
                false,
                false
            ],
            [
                "-webkit-gradient(linear,left 0,left 100%,from(#d9534f),to(#c9302c))"
            ]
        ],
        [
            [
                "background-image",
                false,
                false
            ],
            [
                "-webkit-linear-gradient(top,#d9534f 0,#c9302c 100%)"
            ]
        ],
        [
            [
                "background-image",
                false,
                false
            ],
            [
                "-webkit-linear-gradient(top,#d9534f 0,#c12e2a 100%)"
            ]
        ]
    ]
}
```

**Comment by jakubpawlowicz** [comment]
@rgbohner - d'oh - totally! Fix is already in already in 3.2 branch which I'll release with one more tweak tomorrow morning.

**Comment by rgbohner** [comment]
cools, thanks for the update. love the project

**Comment by jakubpawlowicz** [comment]
Thanks, it means a lot!

The bug is fixed in 3.2.4.
